include/util/guidance/turn_lanes.hpp

```diff
@@ -0,0 +1,95 @@
+#ifndef OSRM_UTIL_GUIDANCE_TURN_LANES_HPP
+#define OSRM_UTIL_GUIDANCE_TURN_LANES_HPP
+
+#include <cstddef>
+#include <cstdint>
+#include <functional>
+#include <vector>
+
+#include "util/typedefs.hpp"
+
+#include <boost/functional/hash.hpp>
+
+namespace osrm
+{
+namespace util
+{
+namespace guidance
+{
+class LaneTupel;
+} // namespace guidance
+} // namespace util
+} // namespace osrm
+
+namespace std
+{
+template <> struct hash<::osrm::util::guidance::LaneTupel>
+{
+    inline std::size_t operator()(const ::osrm::util::guidance::LaneTupel &bearing_class) const;
+};
+} // namespace std
+
+namespace osrm
+{
+namespace util
+{
+namespace guidance
+{
+
+// The mapping of turn lanes can be done using a tupel as described here.
```

Comment:
Keeping track of this so we don't forget about it before merging:

- [ ] remove outdated example or adapt to new representation